From b9627ed74ab1f58be0de075be68d5cd2e96b8750 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Sun, 3 May 2020 23:08:54 +0200
Subject: [PATCH] Use sigset_t instead of __sigset_t on AArch64

This fixes the build with the musl C library. sigset_t is the
documented type exposed by <signal.h>.

This fix comes from Alpine Linux:

  https://git.alpinelinux.org/aports/plain/main/libunwind/fix-aarch64-sigset_t.patch

It was authored by Leo <thinkabit.ukim@gmail.com>.

The issue has been reported upstream at
https://github.com/libunwind/libunwind/issues/176.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 include/libunwind-aarch64.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/libunwind-aarch64.h b/include/libunwind-aarch64.h
index 778b4362..926fbbc0 100644
--- a/include/libunwind-aarch64.h
+++ b/include/libunwind-aarch64.h
@@ -34,6 +34,7 @@ extern "C" {
 #include <inttypes.h>
 #include <stddef.h>
 #include <ucontext.h>
+#include <signal.h>
 
 #define UNW_TARGET      aarch64
 #define UNW_TARGET_AARCH64      1
@@ -192,7 +193,7 @@ typedef struct
 	unsigned long uc_flags;
 	struct ucontext *uc_link;
 	stack_t uc_stack;
-	__sigset_t uc_sigmask;
+	sigset_t uc_sigmask;
 	struct unw_sigcontext uc_mcontext;
   } unw_tdep_context_t;
 
-- 
2.26.2

